mon-pix/app/components/authentication/login-or-register-oidc.js

```diff
@@ -91,14 +97,42 @@ export default class LoginOrRegisterOidcComponent extends Component {
     }
   }
 
+  @action
+  setPassword(event) {
+    this.password = event.target.value;
+  }
+
   get isFormValid() {
     return isEmailValid(this.email) && !isEmpty(this.password);
   }
 
   @action
-  confirmReconciliation() {
+  async login(event) {
+    event.preventDefault();
+    this.loginErrorMessage = null;
+    this.loginError = false;
     if (this.isFormValid) {
-      // todo
+      const identityProvider = IdentityProviders[this.args.identityProviderSlug]?.code;
+      try {
+        const authenticationRequest = this.store.createRecord('user-oidc-authentication-request', {
+          password: this.password,
+          email: this.email,
+          authenticationKey: this.args.authenticationKey,
+          identityProvider,
+        });
+        await authenticationRequest.login();
+        this.args.toggleOidcReconciliation();
+      } catch (error) {
+        this.loginError = true;
+        const status = get(error, 'errors[0].status');
+        if (status === '401') {
+          this.loginErrorMessage = this.intl.t(ERROR_INPUT_MESSAGE_MAP['expiredAuthenticationKey']);
+        } else if (status === '404') {
+          this.loginErrorMessage = this.intl.t(ERROR_INPUT_MESSAGE_MAP['loginUnauthorizedError']);
+        } else {
+          this.loginErrorMessage = this.intl.t(ERROR_INPUT_MESSAGE_MAP['unknownError']);
+        }
```

Comment:
On pourrait créer un objet au niveau du composant 

```js
const errorsMapping = {
  '401': this.intl.t(ERROR_INPUT_MESSAGE_MAP['expiredAuthenticationKey']),
  '404': this.intl.t(ERROR_INPUT_MESSAGE_MAP['loginUnauthorizedError']),
  'unknown': this.intl.t(ERROR_INPUT_MESSAGE_MAP['unknownError']),
};
```

ensuite on pourrait faire ceci

```suggestion
        const status = get(error, 'errors[0].status', 'unknown');
        this.loginErrorMessage = this.errorsMapping[status]
```